packages/wonder-blocks-modal/components/one-column.modal.test.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+import * as React from "react";
+import {shallow} from "enzyme";
+
+import {
+    MediaLayoutContext,
+    MEDIA_DEFAULT_SPEC,
+} from "@khanacademy/wonder-blocks-layout";
+import OneColumnModal, {
+    SmallOneColumnModal,
+    LargeOneColumnModal,
+} from "./one-column-modal.js";
+import {mount, unmountAll} from "../../../utils/testing/mount.js";
+
+describe("OneColumnModal", () => {
+    beforeEach(() => {
+        unmountAll();
+    });
+
+    it("should render SmallOneColumnModal on mobile", () => {
+        // Arrange
+        const wrapper = mount(
+            <MediaLayoutContext.Provider
+                value={{
+                    overrideSize: undefined,
+                    ssrSize: "small",
+                    mediaSpec: MEDIA_DEFAULT_SPEC,
+                }}
+            >
+                <OneColumnModal content="Content" footer="Footer" />
+            </MediaLayoutContext.Provider>,
+        );
+
+        // Act
+        const modal = wrapper.find(SmallOneColumnModal);
+
+        // Assert
+        expect(modal);
+    });
+
+    it("should render LargeOneColumnModal on desktop", () => {
+        // Arrange
+        const wrapper = mount(
+            <MediaLayoutContext.Provider
+                value={{
+                    overrideSize: undefined,
+                    ssrSize: "large",
+                    mediaSpec: MEDIA_DEFAULT_SPEC,
+                }}
+            >
+                <OneColumnModal content="Content" footer="Footer" />
+            </MediaLayoutContext.Provider>,
+        );
+
+        // Act
+        const modal = wrapper.find(LargeOneColumnModal);
+
+        // Assert
+        expect(modal);
+    });
+
+    // We already capture desktop snapshots from auto generated tests
+    describe("mobile", () => {
+        test("with footer", () => {
```

Comment:
Although there's only one test case for this variation right now, it can be useful to use `describe` to layout general scenarios and then have cases within that.

Like `describe("with footer")` then have a test case like `it("should match snapshot")` or  `test("it matches snapshot")`.

The nice thing about that is that we then document the circumstances and the expectation, rather than just the circumstances as in the code as it stands.

What do you think?

Reply:
Sounds good.